### ISSUE TITLE
Getting the openid details from the oauth2 data

### ISSUE THREAD

**Issue by netpointneil** [XeroAPI/xero-php-oauth2#5]
Hi,

Is there any way that you can also retrieve the openid details from the oauth2 successful login response?

The only way I found was to modify League\OAuth2\Client\Provider\AbstractProvider to also add $response['id_token'] to the access token that the function getAccessToken returns.

When we have the id_token, we can perform the following to retrieve the user's email and id that they did agree to share with us when logging into Xero... I found the spec on openid to work out how to get this data out...

```
    $idParts = explode('.', $idToken);
    if (isset($idParts[1])) {
      if ($idPartDecoded = base64_decode($idParts[1])) {
        if ($idPartObject = json_decode($idPartDecoded)) {
          return $idPartObject;
        }
      }
    }

```

Is there any way you can support this feature?  When a user logs in to multiple xero accounts in an application, we need to have a unique identifier.  This is possible through the openid that Xero supports, but isn't yet reflected in this library.

Thanks for consideration.

**Comment by MattR-Api** [comment]
Hi @netpointneil 

Thanks for the suggestion. I'll run it past the Developers to see if it can be included. 

Thanks 

Matt

**Comment by netpointneil** [comment]
Hey, @MattR-Api any update on this?  Really need the openid details made available to applications.

**Comment by SidneyAllen** [comment]
Hi @netpointneil -  i've released version 1.0.4 with some JWT decoding.

Look at the section in the README - https://github.com/XeroAPI/xero-php-oauth2#jwt-decoding-and-signup-with-xero

**Comment by netpointneil** [comment]
Hey, this is great, many thanks for this!

I have found a couple of bugs I think, will raise a new issue.
